src/stirling/bpf_tools/bcc_wrapper.h

```diff
@@ -480,6 +455,191 @@ class BCCWrapper {
+    std::vector<std::pair<K, V>> r;
+
+    // This is a user space managed map: we can iterate over the shadow keys.
+    for (const auto& k : shadow_keys_) {
+      auto s = GetValue(k);
+      const auto v = s.ConsumeValueOrDie();
+      r.push_back({k, v});
+      if (clear_table) {
+        PX_UNUSED(underlying_->remove_value(k));
+      }
+    }
+    if (clear_table) {
+      shadow_keys_.clear();
+    }
+    return r;
+  }
+
+ private:
+  WrappedBCCMap(bpf_tools::BCCWrapper* bcc, const std::string& name) : name_(name) {
+    underlying_ = std::make_unique<U>(bcc->BPF().get_hash_table<K, V>(name_));
+  }
+
+  const std::string name_;
+  char const* const err_msg_ = "BPF failed to $0 value for map: $1. $2.";
+  std::unique_ptr<U> underlying_;
+  absl::flat_hash_set<K> shadow_keys_;
+};
+
+template <typename T>
+class WrappedBCCPerCPUArrayTable {
+ public:
+  using U = ebpf::BPFPercpuArrayTable<T>;
+
+  static std::unique_ptr<WrappedBCCPerCPUArrayTable> Create(bpf_tools::BCCWrapper* bcc,
+                                                            const std::string& name) {
+    return std::unique_ptr<WrappedBCCPerCPUArrayTable>(new WrappedBCCPerCPUArrayTable(bcc, name));
+  }
+
+  Status SetValues(const int idx, const T& value) {
+    std::vector<T> values(bpf_tools::BCCWrapper::kCPUCount, value);
+    ebpf::StatusTuple s = underlying_->update_value(idx, values);
+    if (!s.ok()) {
+      char const* const err_msg_ = "BPF failed to $0 value for per cpu array: $1, index: $2. $3.";
+      return error::Internal(absl::Substitute(err_msg_, "set", name_, idx, s.msg()));
+    }
+    return Status::OK();
+  }
+
+ private:
+  WrappedBCCPerCPUArrayTable(bpf_tools::BCCWrapper* bcc, const std::string& name) : name_(name) {
+    underlying_ = std::make_unique<U>(bcc->BPF().get_percpu_array_table<T>(name_));
+  }
+
+  const std::string name_;
+  std::unique_ptr<U> underlying_;
+};
+
+class WrappedBCCStackTable {
+ public:
+  using U = ebpf::BPFStackTable;
+
+  static std::unique_ptr<WrappedBCCStackTable> Create(bpf_tools::BCCWrapper* bcc,
+                                                      const std::string& name) {
+    return std::unique_ptr<WrappedBCCStackTable>(new WrappedBCCStackTable(bcc, name));
+  }
+
+  std::vector<uintptr_t> GetStackAddr(const int stack_id, const bool clear_stack_id) {
+    return underlying_->get_stack_addr(stack_id, clear_stack_id);
+  }
+
+  std::string GetAddrSymbol(const uintptr_t addr, const int pid) {
+    return underlying_->get_addr_symbol(addr, pid);
+  }
+
+  void ClearStackID(const int stack_id) { underlying_->clear_stack_id(stack_id); }
+
+  U* RawPtr() { return underlying_.get(); }
+
+ private:
+  WrappedBCCStackTable(bpf_tools::BCCWrapper* bcc, const std::string& name) : name_(name) {
+    underlying_ = std::make_unique<U>(bcc->BPF().get_stack_table(name_));
+  }
+
+  const std::string name_;
+  std::unique_ptr<U> underlying_;
+};
+
+// ebpf::BPFStackTable GetStackTable(const std::string& table_name) {
```

Comment:
Any reason this is commented out? Should it be removed?

Reply:
Yikes, that should be removed.